Compile instructions

Is there any instructions for compiling the bot as I would like to run it on a raspberry pi with an ARM processor.

I deleted the file and tried compiling and got the following error:
 ```
error[E0432]: unresolved import `core::task::Context`                                                          
  --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:21:18  
   | 
21 | use core::task::{Context, Poll};                                                                          
   |                  ^^^^^^^ no `Context` in `task`                                                    
     
error[E0658]: use of unstable library feature 'alloc': this library is unlikely to be stabilized in its current form or name (see issue #27783)
  --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:17:1   
   | 
17 | extern crate alloc;                                                                                       
   | ^^^^^^^^^^^^^^^^^^^                                                                                
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:21:27  
   | 
21 | use core::task::{Context, Poll};
   |                           ^^^^                                                                     
                            
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:66:66
   |                        
66 |     fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>>;           
   |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^     
                        
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:101:66
    |                           
101 |     fn poll_flush(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>>;          
    |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^    
                              
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:114:66
    |                              
114 |     fn poll_close(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>>;          
    |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^    
                               
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:120:70
    |                      
120 |     fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {     
    |                                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                       
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:128:70
    |                                
128 |     fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {     
    |                                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                           
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:132:70
    |                           
132 |     fn poll_close(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {     
    |                                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:144:66
    |
144 |     fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {                                         
    |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^    
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:152:66
    |
152 |     fn poll_flush(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {         
    |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^    
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:156:66
   |     fn poll_close(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {         
    |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  
                            
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:169:69
    |                   
169 |         fn poll_ready(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {      
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
                                
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:170:13
    |                         
170 |             Poll::Ready(Ok(()))                                                                          
    |             ^^^^^^^^^^^                                                                           
                                   
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:179:69
    |                          
179 |         fn poll_flush(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {      
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
                           
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:180:13
    |                  
180 |             Poll::Ready(Ok(()))                                                                          
    |             ^^^^^^^^^^^                                                                           
                                     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:183:69
    |                      
183 |         fn poll_close(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {      
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
                                
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:184:13
    |
184 |             Poll::Ready(Ok(()))                                                                          
    |             ^^^^^^^^^^^                                                                           
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)                                
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:191:69
    |
191 |         fn poll_ready(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {      
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:192:13
    |
192 |             Poll::Ready(Ok(()))                                                                          
    |             ^^^^^^^^^^^
    error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:201:69
    |                       
201 |         fn poll_flush(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {      
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
                        
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:202:13
    |                           
202 |             Poll::Ready(Ok(()))                                                                          
    |             ^^^^^^^^^^^                                                                           
                              
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:205:69
    |                              
205 |         fn poll_close(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {      
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
                               
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:206:13
    |                      
206 |             Poll::Ready(Ok(()))                                                                          
    |             ^^^^^^^^^^^                                                                           
                       
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:213:74
    |                                
213 |         fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> { 
    |                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                           
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:221:74
    |                           
221 |         fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> { 
    |                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     
error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.4/src/lib.rs:225:74
    |
225 |         fn poll_close(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {                                 
    |                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     
error: aborting due to 27 previous errors                                                                      
                                                                                                        
Some errors occurred: E0432, E0658.
For more information about an error, try `rustc --explain E0432`.                                              
error: Could not compile `futures-sink`.                                                                
warning: build failed, waiting for other jobs to finish...
error: build failed
```
I am using the version in the debian repo:
```
Package: rustc
Version: 1.34.2+dfsg1-1+rpi1
```

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 # pokebot
 
 ```
-pokebot 0.1.1
+pokebot 0.2.0
 Jokler <jokler@protonmail.com>
 
 USAGE:
@@ -14,14 +14,15 @@ FLAGS:
     -v, --verbose    Print the content of all packets
 
 OPTIONS:
-    -a, --address <address>                     The address of the server to connect to
-    -g, --generate-identities <gen_id_count>    Generate 'count' identities
-    -d, --master_channel <master_channel>       The channel the master bot should connect to
+    -a, --address <address>                         The address of the server to connect to
+    -g, --generate-identities <gen_id_count>        Generate 'count' identities
+    -d, --master_channel <master_channel>           The channel the master bot should connect to
+    -w, --increase-security-level <wanted_level>    Increases the security level of all identities in the config file
 
 ARGS:
     <config_path>    Configuration file [default: config.toml]
 ```
-# Usage
+## Usage
 
  1. Poke the main bot.
  2. Once the secondary bot joins your channel, type !help for a list of commands.
@@ -40,3 +41,22 @@ ARGS:
     stop      Stops audio playback
     volume    Changes the volume to the specified value
  ```
+
+## Compiling
+
+1. Make sure the following are installed
+    * cargo + rustc 1.42 or later
+    * `gstreamer` development libraries which should be `libgstreamer-dev` and `libgstreamer-plugins-base-dev`
+
+2. Clone the source with `git`:
+    ```sh
+    $ git clone https://github.com/Mavulp/pokebot.git
+    $ cd pokebot
+    ```
+
+3. Building the binary
+    ```sh
+    $ cargo build --release
+    ```
+
+    This creates the binary under `target/release/`.

